### PR TITLE
Support for Parse Schema

### PR DESCRIPTION
**Pull request by dplewis**
Pending: https://github.com/parse-community/parse-server/pull/4240

Adds CRUD support for new Parse.Schema class.

### CODE REVIEW

File: src/ParseSchema.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+ * <p>A Parse.Schema object is for handling schema data from Parse.
+ * All the schemas methods require master key.</p>
+ */
+export default class ParseSchema {
+  className: string;
+  _fields: { [key: string]: mixed };
+  _indexes: { [key: string]: mixed };
+
+  constructor(className: ?string) {
+    if (typeof className === 'string') {
```
Comment (flovilmart) [inline src/ParseSchema.js]: Can we get jsdocs in all methods please? :)

**Comment by flovilmart** [comment]
@dplewis, have a look at the overhaul I had to do for the docs to be compatible with JSDoc 3.0. That will help so I don’t have to re-do it :p

**Comment by dplewis** [comment]
@flovilmart I'll add the documentation. Can you check to ensure the tests pass against parse-server 2.7.0 (travis)?

**Comment by flovilmart** [comment]
Yes all good, I'll bump travis in my PR too :)

**Comment by dplewis** [comment]
@flovilmart I always wondered are these tests still used? https://github.com/parse-community/Parse-SDK-JS/tree/master/src/__tests__

**Comment by flovilmart** [comment]
Yes they are, those are the jest tests :(

**Comment by flovilmart** [comment]
But they don’t report coverage, each file is unit tested by Jest, integration tests are run later and don’t produce coverage reports.

**Comment by dplewis** [comment]
@flovilmart Can you look this over?

**Comment by dplewis** [comment]
@flovilmart jest or no? :(

**Comment by flovilmart** [comment]
Looking great man!

**Pull request merged**